pkg/nfspv/nfsvolume.go

```diff
@@ -0,0 +1,302 @@
+/*
+Copyright © 2021 The MayaData Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfspv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/mayadata-io/volume-events-exporter/pkg/collectorinterface"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+const (
+	OpenEBSNFSLabelKey = "nfs.openebs.io/volume"
+)
+
+// nfsVolume will implement necessary methods
+// required to satisfy collector interface
+type nfsVolume struct {
+	clientset kubernetes.Interface
+
+	pvcLister corev1listers.PersistentVolumeClaimLister
+
+	pvLister           corev1listers.PersistentVolumeLister
+	pvObj              *corev1.PersistentVolume
+	nfsServerNamespace string
+	annotationPrefix   string
+}
+
+func NewNFSVolume(
+	clientset kubernetes.Interface,
+	pvcLister corev1listers.PersistentVolumeClaimLister,
+	pvLister corev1listers.PersistentVolumeLister,
+	pvObj *corev1.PersistentVolume,
+	nfsServerNamespace string) collectorinterface.VolumeEventCollector {
+	return &nfsVolume{
+		clientset:          clientset,
+		pvcLister:          pvcLister,
+		pvLister:           pvLister,
+		pvObj:              pvObj,
+		nfsServerNamespace: nfsServerNamespace,
+		annotationPrefix:   "nfs.",
+	}
+}
+
+func (n *nfsVolume) CollectCreateEventData() (string, error) {
+
+	volumeData, err := n.getVolumeData()
+	if err != nil {
+		return "", err
+	}
+
+	if volumeData.NFSPVC != nil {
+		volumeData.NFSPVC.DeletionTimestamp = nil
+		volumeData.NFSPVC.DeletionGracePeriodSeconds = nil
+	}
+
+	volumeData.NFSPV.DeletionTimestamp = nil
```

Comment:
nit: better to add a comment here stating, to send create and delete event, if create event missed, we are setting this values to nil....

Reply:
Added comment before starting the function